proposals/v3/grpc.md

```diff
@@ -0,0 +1,349 @@
+| *sortIndex* | int32 | A 1-based sort ordinal for the device. This will help determine where the device shows up in the scan. |
+| *capabilities* | [V3DeviceCapability](#v3devicecapability) |  |
+| *output* | repeated [V3Output](#v3deviceoutput) | The reading outputs that the device can generate on read. |
+
+
+#### V3DeviceCapability
+Specifies the capabilities that a device exposes via Synse.
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| *mode* | string | The capability mode of the device ("ro", "wo", "rw"). |
+| *write* | [V3WriteCapability](#v3writecapability) | The write capabilities of the device. |
+
+
+#### V3DeviceOutput
+Specifies the output types for a device's readings.
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| *name* | string | The name of the device output. This can be namespaced. |
+| *type* | string | The type of the output. This is the last element of the namespaced output name. |
+| *precision* | int | The decimal precision of the output. A precision of 0 (default) means no precision is applied. |
+| *scalingFactor* | double | The factor to multiply the reading result returned from the device. This can be positive, negative, whole, or decimal. |
+| *unit* | [V3OutputUnit](#v3outputunit) | The unit of measure for the reading output. |
+
+
+#### V3DeviceSelector
+A selector for choosing devices for various operations.
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| *tags* | repeated [V3Tag](#v3tag) | A collection of all the tags to be used as selectors. |
+
+
+#### V3Health
+Health status for the plugin.
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| *timestamp* | string | RFC3339 timestamp of the time when the health was checked. |
+| *status* | enum HealthStatus | The overall status (one of: unknown, ok, failing). |
+| *checks* | repeated [V3HealthCheck](#v3healthcheck) | All of the health checks which make up the overall health of a plugin. |
+
+
+#### V3HealthCheck
+Health check status for a plugin.
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| *name* | string | The name of the health check. |
+| *status* | enum HealthStatus | The status of the health check. |
+| *message* | string | Any additional information associated with the health check. |
+| *timestamp* | string | RFC3339 timestamp at which the check was last completed. |
+| *type* | string | The type of health check. The different kinds of health check are defined in the SDK. |
+
+
+#### V3Metadata
+Plugin metadata. This information static information about the plugin.
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| *name* | string | The name of the plugin. |
+| *maintainer* | string | The maintainer of the plugin. |
+| *tag* | string | The normalized tag for the plugin metainfo. |
+| *description* | string | A brief description of the plugin. |
+| *vcs* | string | A link to the plugin's VCS repo. |
+
+
+#### V3OutputUnit
```

Comment:
true. it would be nice to have this natively supported. i'll spend some time thinking through how we may be able to get this done. even if this doesn't make it into v3.0, it would be good to have a sense on how to approach this for v3.X

Reply:
I brought that up even though it's tangential to this review. Different discussion, but if we are changing everything it should be considered.

Reply:
I started adding some notes in e4a4cb6 thinking through how this can fit in. Not complete yet.